pkg/controller/job/job_controller_test.go

```diff
@@ -413,6 +414,39 @@ func TestControllerSyncJob(t *testing.T) {
 			expectedCreatedIndexes: sets.NewInt(0, 1),
 			indexedJobEnabled:      true,
 		},
+		"indexed job completed": {
+			parallelism:    2,
+			completions:    3,
+			backoffLimit:   6,
+			completionMode: batch.IndexedCompletion,
+			jobKeyForget:   true,
+			podsWithIndexes: []indexPhase{
+				{"0", v1.PodSucceeded},
+				{"1", v1.PodFailed},
+				{"1", v1.PodSucceeded},
+				{"2", v1.PodSucceeded},
+			},
+			expectedSucceeded:     3,
+			expectedFailed:        1,
+			expectedCompletedIdxs: "0-2",
+			expectedCondition:     &jobConditionComplete,
+			indexedJobEnabled:     true,
+		},
+		"indexed job repeated completed index": {
+			parallelism:    2,
+			completions:    3,
+			backoffLimit:   6,
+			completionMode: batch.IndexedCompletion,
+			jobKeyForget:   true,
+			podsWithIndexes: []indexPhase{
+				{"0", v1.PodSucceeded},
+				{"1", v1.PodSucceeded},
+				{"1", v1.PodSucceeded},
+			},
+			expectedSucceeded:     3,
+			expectedCompletedIdxs: "0,1",
```

Comment:
I'll leave a question here from slack for discussion, this situation seems unusual, and I'm trying to figure out if with the current controller implementation it is possible to have duplicate completed indexes? Should we still count successfully completed as including those duplicates, or rather they should match completed indexes?

Reply:
The semantics for indexed job should be: each index is executed it least once. Ideally it should be at most one too, but I don't know if we can guarantee that. So having this being checked is good, it may not happen, but at least we have the logic that guarantee the at least once semantics.

Reply:
sgtm

Reply:
correct, it shouldn't happen. But it might happen when the controller restarts. And in general the controller will try to terminate duplicated active pods, but if the controller is down, all pods might finish.